### ISSUE TITLE
Duplicate component in view

### ISSUE THREAD

**Issue by bodenloszt** [storybookjs/storybook#13859]
**Describe the bug**
I was playing around with the Component.args functionality. After running Storybook, when I change the e.g. children argument from the controls panel, for each letter written, it renders another component into my view.

**Expected behavior**
It should replace the previous button children.

**Screenshots**
<img width="939" alt="Untitled" src="https://user-images.githubusercontent.com/77001547/107489013-ce90f080-6b90-11eb-9a68-434b834c95d9.png">

**Code snippets**
```
import { h } from "preact";
import ButtonComponent from "./button";

export default {
    title: "Components/ButtonComponents/Button",
    component: ButtonComponent
};

const Template = args => <ButtonComponent {...args} />;

export const FirstButton = Template.bind({});
FirstButton.args = {
    type: "button",
    href: "",
    target: "_blank",
    onClick: null,
    children: "First Button"
};
```

**System**
```
 System:
    OS: Windows 10 10.0.18363
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
  Binaries:
    Node: 12.18.2 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.5 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 87.0.4280.141
    Edge: Spartan (44.18362.449.0)
  npmPackages:
    @storybook/addon-actions: ^6.1.14 => 6.1.14
    @storybook/addon-decorator: ^6.1.17 => 6.1.17
    @storybook/addon-essentials: ^6.1.14 => 6.1.14
    @storybook/addon-links: ^6.1.14 => 6.1.14
    @storybook/preact: ^6.1.14 => 6.1.14
```

**Comment by shilman** [comment]
Do you have a repro repo you can share?

**Comment by bodenloszt** [comment]
> Do you have a repro repo you can share?

Unfortunately I cannot, but if it helps it is still only this component I posted about here: https://github.com/storybookjs/storybook/issues/13635

**Comment by timohermans** [comment]
@bodenloszt @shilman  this has been fixed in [6.3.0.rc-1 See ](https://github.com/storybookjs/storybook/commit/202bfa548be19dfe807bd1384e48e2eff0c79449)

**Comment by shilman** [comment]
great, thanks for letting me know. closing!
